pkg/subctl/cmd/cloud/cleanup/gcp.go

```diff
@@ -20,38 +20,50 @@ package cleanup
 
 import (
 	"github.com/spf13/cobra"
-	"github.com/submariner-io/admiral/pkg/reporter"
-	"github.com/submariner-io/cloud-prepare/pkg/api"
 	"github.com/submariner-io/submariner-operator/internal/cli"
 	"github.com/submariner-io/submariner-operator/internal/exit"
+	"github.com/submariner-io/submariner-operator/internal/restconfig"
+	"github.com/submariner-io/submariner-operator/pkg/cloud/cleanup"
+	cloudgcp "github.com/submariner-io/submariner-operator/pkg/cloud/gcp"
 	"github.com/submariner-io/submariner-operator/pkg/subctl/cmd/cloud/gcp"
+	"github.com/submariner-io/submariner-operator/pkg/subctl/cmd/cloud/utils"
 )
 
+var gcpConfig cloudgcp.Config
+
 // newGCPCleanupCommand returns a new cobra.Command used to prepare a cloud infrastructure.
-func newGCPCleanupCommand() *cobra.Command {
+func newGCPCleanupCommand(restConfigProducer restconfig.Producer) *cobra.Command {
 	cmd := &cobra.Command{
 		Use:   "gcp",
 		Short: "Clean up a GCP cloud",
 		Long:  "This command cleans up an installer-provisioned infrastructure (IPI) on GCP-based cloud after Submariner uninstallation.",
-		Run:   cleanupGCP,
-	}
+		Run: func(cmd *cobra.Command, args []string) {
+			status := cli.NewReporter()
 
-	gcp.AddGCPFlags(cmd)
+			var err error
+			if gcpConfig.OcpMetadataFile != "" {
+				gcpConfig.InfraID, gcpConfig.Region, gcpConfig.ProjectID, err = cloudgcp.ReadFromFile(gcpConfig.OcpMetadataFile)
+				exit.OnErrorWithMessage(err, "Failed to read GCP Cluster information from OCP metadata file")
+			} else {
+				utils.ExpectFlag(utils.InfraIDFlag, gcpConfig.InfraID)
+				utils.ExpectFlag(utils.RegionFlag, gcpConfig.Region)
+				utils.ExpectFlag(utils.ProjectIDFlag, gcpConfig.Region)
+			}
 
-	return cmd
-}
+			gcpConfig.GWInstanceType = ""
+			gcpConfig.DedicatedGateway = false
```

Comment:
Why is this necessary?

Reply:
Well, `cleanup` passes empty `GW instance type` and false `dedicated GWs` as arguments to `gcp,RunOn` (along with infra-id, region, etc...). I understand that the default values for string is empty and bool is false. I just mentioned that explicitly here.